datarobot_batch_scoring/writer.py

```diff
@@ -410,53 +416,108 @@ def format_result_data(self, result, batch, pred_name):
                            key=operator.itemgetter('row_id'))]
             out_fields = ['row_id', pred_name if pred_name else '']
         else:
-            ValueError('task "{}" not supported'
-                       ''.format(result['task']))
-        return (out_fields, pred)
+            raise ValueError('task "{}" not supported'
+                             ''.format(result['task']))
 
-    def sigterm_handler(self, _signo, _stack_frame):
```

Comment:
if you sigkill the master - nothing is going to help you.. Removing everything, but supervision from master was done exactly for this reason - master is not supposed to die in any case.

master itself is looking for all children and triggers proper shutdown if any of them die.

if you sigkill any of workers - it will trigger "worker diead without reporting success" in master and anything will be shutdown properly.